packages/@lwc/ssr-compiler/src/compile-template/transformers/component.ts

```diff
@@ -35,7 +35,11 @@ const bYieldFromChildGenerator = esTemplateWithYield`
         const childAttrs = ${/* child attrs */ is.objectExpression};
         const slottedContent = {
             light: Object.create(null),
-            shadow: async function* () {
+
+            // The 'instance' variable is shadowed here so that a contextful relationship
+            // is established between components rendered in slotted content & the "parent"
+            // component that contains the <slot>.
+            shadow: async function* (instance) {
```

Comment:
This 1) conflicts with the first parameter passed in for scoped slots, and 2) should be applied to light DOM slots and scoped slots as well.

I suggest that the first argument should always be `instance` and the second argument the optional scoped-slot-bound-variable.

Reply:
https://github.com/salesforce/lwc/issues/4829